### PR TITLE
9KelvinN

### PR DESCRIPTION
**Pull request by 9KelvinN**
The puzzle scraper gathering the problems from chess.com and all their attributes

### CODE REVIEW

File: src/PuzzleScraper.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self.ID = attributes[0].text.strip()
+        self.Rating = attributes[1].text.strip()
+        self.Attempts = attributes[2].text.strip()
+        self.PassPercent = attributes[3].text.strip()
+        self.NumMoves = attributes[4].text.strip()
+        self.AvgTime = attributes[5].text.strip()
+        self.PlayAsColor = attributes[6].text.strip()
+#6720 pages in total
+problems=[]
+for pagenum in range(1,4):
```
Comment (thisistrivial) [inline src/PuzzleScraper.py:17]: Is it just 4 pages?

File: src/PuzzleScraper.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    for problemhtml in problemshtml:
+        problemhtml.get('v-board-popover')
+        try:
+            fen = re.search("fen: '(.+?) ", problemhtml.get('v-board-popover')).group(1)
+        except AttributeError:
+            fen =''
+        #The attributes are id, Rating, Attempts, PassPercent, NumMoves, AvgTime, PlayAsColor
+        attributes = problemhtml.find_all('a')
+        problems.append(problem(fen,attributes))
+#Test
```
Comment (thisistrivial) [inline src/PuzzleScraper.py:31]: Probably best to leave test cases out of `src`. You could instead add the code to a summary / comment on the pull request page

File: src/PuzzleScraper.py (hunk 1 above)
Comment (9KelvinN) [reply src/PuzzleScraper.py:17]: No, it's that int should be 6721. I'll make a new commit fixing that and the data type for the ID and rating
```suggestion
for pagenum in range(1,6721):

```

**Comment by 9KelvinN** [comment]
> Separate between local and upstream source
So I should just create a separate source folder?

**Pull request merged**